Sources/CMakeLists.txt

```diff
@@ -12,12 +12,29 @@
 ##
 ##===----------------------------------------------------------------------===##
 
-add_subdirectory(_FoundationCShims)
-
-# Disable the macro build on Windows until we can correctly build it for the host architecture
-if(NOT CMAKE_SYSTEM_NAME STREQUAL Windows)
-	add_subdirectory(FoundationMacros)
+include(ExternalProject)
+if(CMAKE_HOST_WIN32)
+  set(_SwiftFlags -DCMAKE_Swift_FLAGS="-use-ld=lld")
+else()
+  set(_SwiftFlags)
+endif()
+ExternalProject_Add(FoundationMacros
+  SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/FoundationMacros"
+  PREFIX "${CMAKE_BINARY_DIR}/_deps"
+  BINARY_DIR "macros"
+  CMAKE_ARGS
+    -DCMAKE_MAKE_PROGRAM=${CMAKE_MAKE_PROGRAM}
+    ${_SwiftFlags}
+  INSTALL_COMMAND "")
+ExternalProject_Get_Property(FoundationMacros BINARY_DIR)
+if(CMAKE_HOST_WIN32)
+  set(_SwiftFoundation_PredicateMacro "${BINARY_DIR}/FoundationMacros.exe#PredicateMacro")
+  set(_SwiftFoundation_ExpressionMacro "${BINARY_DIR}/FoundationMacros.exe#ExpressionMacro")
+else()
+  set(_SwiftFoundation_PredicateMacro "${BINARY_DIR}/FoundationMacros#PredicateMacro")
```

Comment:
It doesn't look like these values are used anywhere, is that intentional? Also it looks like it's listing these as executables but the macros build as a library (like the Observation / other macros produced by the stdlib) so I don't think this are quite accurate - I'd expect these to look like `libFoundationMacros.so` and `FoundationMacros.dll`

Reply:
Yes, this is intentional - the macros are not currently used in the CMake build AFAICT. The default behaviour for the macros is to build as executables, the in-process (library) model is something that is still being implemented. Supporting that as an option would add more complexity.

Reply:
@bnbarham remind me - I think we discussed that the macro should be a library rather than an executable on linux (am I remembering that right?). Did we have a reason for that? If so, should we update this to account for that difference (an executable on Windows but still building as an `.so` on linux)?

Reply:
> The default behaviour for the macros is to build as executables

For external macros that's true, but for macros included in the toolchain it's better if they're a library linking to the `lib/swift/host` swift-syntax which is then loaded into `swift-plugin-server`. They could theoretically be executables, though I'm not sure we have any search paths for executable macros in the toolchain by default (CC @rintaro in case I'm wrong)

> the in-process (library) model is something that is still being implemented

In-process already exists, the work @rintaro has been doing is to allow a separate swift-syntax to be used for those libraries compared to the compiler. In any case, that isn't what would be being used here anyway.